wrappers/java/src/main/java/org/hyperledger/indy/sdk/IndyException.java

```diff
@@ -1,40 +1,133 @@
+import org.hyperledger.indy.sdk.pool.PoolConfigNotCreatedException;
+import org.hyperledger.indy.sdk.pool.PoolLedgerConfigExistsException;
+import org.hyperledger.indy.sdk.pool.PoolLedgerTerminatedException;
+import org.hyperledger.indy.sdk.signus.UnknownCryptoException;
+import org.hyperledger.indy.sdk.wallet.DuplicateWalletTypeException;
+import org.hyperledger.indy.sdk.wallet.UnknownWalletTypeException;
+import org.hyperledger.indy.sdk.wallet.WalletAlreadyOpenedException;
+import org.hyperledger.indy.sdk.wallet.WalletClosedException;
+import org.hyperledger.indy.sdk.wallet.WalletExistsException;
+import org.hyperledger.indy.sdk.wallet.WalletValueNotFoundException;
+import org.hyperledger.indy.sdk.wallet.WrongWalletForPoolException;
+
 /**
- * Indy specific exception.
+ * Thrown when an Indy specific error has occurred.
  */
 public class IndyException extends Exception {
 
 	private static final long serialVersionUID = 2650355290834266477L;
-
-	private ErrorCode errorCode;
+	private int sdkErrorCode;
 
 	/**
 	 * Initializes a new IndyException with the specified message.
 	 * 
 	 * @param message The message for the exception.
 	 */
-	public IndyException(String message) {
-
+	protected IndyException(String message, int sdkErrorCode) {
 		super(message);
+		this.sdkErrorCode = sdkErrorCode;
 	}
 
 	/**
-	 * Initializes a new IndyException using the specified ErrorCode.
+	 * Gets the SDK error code for the exception.
 	 * 
-	 * @param errorCode The error code for the exception.
+	 * @return The SDK error code used to construct the exception.
 	 */
-	public IndyException(ErrorCode errorCode) {
-		this(String.format("%s: %d", errorCode.name(), errorCode.value()));
-		this.errorCode = errorCode;
+	public int getSdkErrorCode() {
+		return sdkErrorCode;
 	}
-
+	
 	/**
-	 * Gets the ErrorCode for the exception.
+	 * Initializes a new IndyException using the specified SDK error code.
 	 * 
-	 * @return The ErrorCode used to construct the exception.
+	 * @param sdkErrorCode The SDK error code to construct the exception from.
 	 */
-	public ErrorCode getErrorCode() {
-		return errorCode;
+	public static IndyException fromSdkError(int sdkErrorCode) {
+		
+		ErrorCode errorCode = ErrorCode.valueOf(sdkErrorCode);
+		
+		switch(errorCode){
+			case CommonInvalidParam1:
+			case CommonInvalidParam2:
+			case CommonInvalidParam3:
+			case CommonInvalidParam4:
+			case CommonInvalidParam5:
+			case CommonInvalidParam6:
+			case CommonInvalidParam7:
+			case CommonInvalidParam8:
+			case CommonInvalidParam9:
+			case CommonInvalidParam10:
+			case CommonInvalidParam11:
+			case CommonInvalidParam12:
+				return new InvalidParameterException(sdkErrorCode);
+			case CommonInvalidState:
+				return new InvalidStateException();
+			case CommonInvalidStructure:
+				return new InvalidStructureException();
+			case CommonIOError:
+				return new IOException();
+			case WalletInvalidHandle:
+				return new WalletClosedException();
```

Comment:
WalletInvalidHandleException is not really a clear description either - the developer never sees the handle as they only work with Wallet instances.  Perhaps there's a better term we could use?

Reply:
Thanks for note, java wrapper API is actually handle-less.
As for *current implementation* of wrapper for wallet it's actually only WalletClosed...
But if we modify something it may add another reasons. So both WalletClosed and WalletInvalidHandle seems inaccuracy.
May be InvalidWallet or WalletInvalidState are more applicable terms. I haven't good idea for this exception. @vimmerru?

Reply:
@srottem, we discuss with @vimmerru and choose `InvalidWallet` and `InvalidPool`. Please use it, if you will not find better names.

Reply:
They work for me.